### PR TITLE
Actually support plugins field in config

### PR DESCRIPTION
**Pull request by taion**

### CODE REVIEW

**Comment by taion** [comment]
Sec, this only squelches the warning. It doesn't actually work.

**Comment by taion** [comment]
Still one more bug to fix here

**Comment by taion** [comment]
Good to go now. cc @azz

This fixes the hidden README example for prettier-python. Not sure what the test coverage policy here looks like as I didn't see any for #3536.

File: src/cli/util.js [hunk 1]
```diff
@@ -619,7 +622,10 @@ function normalizeConfig(type, rawConfig, options) {
       return;
     }
 
-    const option = constant.detailedOptionMap[key];
+    let option = constant.detailedOptionMap[key];
+    if (type === "api" && option === undefined) {
```
Comment (taion) [inline src/cli/util.js]: Otherwise we barf on unrecognized "plugins" in config file

File: src/cli/util.js [hunk 2]
```diff
@@ -37,9 +37,12 @@ function getOptions(argv) {
   );
 }
 
-function dashifyObject(object) {
+function cliifyOptions(object) {
   return Object.keys(object || {}).reduce((output, key) => {
-    output[dashify(key)] = object[key];
+    const apiOption = constant.apiDetailedOptionMap[key];
+    const cliKey = apiOption ? apiOption.name : key;
```
Comment (taion) [inline src/cli/util.js]: Need to map "plugins" to "plugin", else it always gets clobbered (even if `--plugin` isn't specified on CLI)

File: src/main/options.js [hunk 3]
```diff
@@ -44,6 +44,8 @@ function normalize(options) {
   const normalized = Object.assign({}, options || {});
   const filepath = normalized.filepath;
 
+  normalized.plugins = loadPlugins(normalized);
```
Comment (taion) [inline src/main/options.js]: Need to load plugins first so `getSupportInfo` below can see Python

**Comment by azz** [comment]
The original or didn't have tests because it was largely just a refactor. We should start adding integration tests for this kind of thing.

**Comment by taion** [comment]
What do you think of adding an issue to track that and tackling the testing separately?

On master right now, the `plugins` field in config files doesn't really work at all, and parser inference for languages from plugins also doesn't work.

It's not exactly a blocker, but it makes testing prettier-python take a bit more effort than it needs to.

File: src/cli/constant.js [hunk 4]
```diff
@@ -443,10 +443,19 @@ const detailedOptionMap = detailedOptions.reduce(
   {}
 );
 
+const apiDetailedOptionMap = detailedOptions.reduce(
+  (current, option) =>
+    option.forwardToApi && option.forwardToApi !== option.name
```
Comment (azz) [inline src/cli/constant.js]: Isn't forwardToApi `true` in a lot of places?

Comment (taion) [reply src/cli/constant.js]: At this point they're already normalized to the camelcased strings per https://github.com/taion/prettier/blob/70313e67950edb34a8cab0911c5b3ab1a90b7db6/src/cli/constant.js#L425

**Pull request merged**